src/Ilios/CliBundle/Command/SyncAllUsersCommand.php

```diff
@@ -0,0 +1,260 @@
+                    continue;
+                }
+                $user = $users[0];
+
+                $update = false;
+                $output->writeln(
+                    '<info>[I] Comparing User #' . $user->getId() . ' ' .
+                    $user->getFirstAndLastName() . ' (' . $user->getEmail() . ') ' .
+                    'to directory user by Campus ID ' . $user->getCampusId() . '</info>'
+                );
+                if (!$this->validateDirectoryRecord($recordArray, $output)) {
+                    $user->setExamined(true);
+                    $this->userManager->updateUser($user, false);
+                    //don't do anything else with invalid directory data
+                    continue;
+                }
+                if ($user->getFirstName() != $recordArray['firstName']) {
+                    $update = true;
+                    $output->writeln(
+                        '  <comment>[I] Updating first name from "' . $user->getFirstName() .
+                        '" to "' . $recordArray['firstName'] . '"</comment>'
+                    );
+                    $user->setFirstName($recordArray['firstName']);
+                }
+                if ($user->getLastName() != $recordArray['lastName']) {
+                    $update = true;
+                    $output->writeln(
+                        '  <comment>[I] Updating last name from "' . $user->getLastName() .
+                        '" to "' . $recordArray['lastName'] . '"</comment>'
+                    );
+                    $user->setLastName($recordArray['lastName']);
+                }
+                if ($user->getPhone() != $recordArray['telephoneNumber']) {
+                    $update = true;
+                    $output->writeln(
+                        '  <comment>[I] Updating phone number from "' . $user->getPhone() .
+                        '" to "' . $recordArray['telephoneNumber'] . '"</comment>'
+                    );
+                    $user->setPhone($recordArray['telephoneNumber']);
+                }
+                if ($user->getEmail() != $recordArray['email']) {
+                    if (strtolower($user->getEmail()) == strtolower($recordArray['email'])) {
+                        $update = true;
+                        $output->writeln(
+                            '  <comment>[I] Updating email from "' . $user->getEmail() .
+                            '" to "' . $recordArray['email'] . '" since the only difference was the case.</comment>'
+                        );
+                        $user->setEmail($recordArray['email']);
+                    } else {
+                        $output->writeln(
+                            '  <comment>[I] Email address "' . $user->getEmail() .
+                            '" differs from "' . $recordArray['email'] . '" logging for further action.</comment>'
+                        );
+                        $update = $this->pendingUserUpdateManager->createPendingUserUpdate();
```

Comment:
not sure on this one here. from what i can see in the code above/below, `$update` is a flag to indicate that the existing user record got updated. doesn't seem to be the case here. the conditional check below still resolves to `true`, and the `$updated` counter gets incremented. please rename variable to `$pendingUpdate` or the likes if applicable.